### PR TITLE
[CI] Automate release process

### PR DESCRIPTION
**Pull request by v1v**
### What

Release automation from the `master` branch, with the below support:
* UI based, so it's required to be login in the CI and trigger a build manually with the `release` parameter.
* Email/slack notifications when a new release is triggered.
* Input approval to confirm the release should be out with 12 hours waiting for.
* Release automation with 2FA.

In addition, I updated the docs regarding the release process with the CI.

### Issue

Closes https://github.com/elastic/synthetics/issues/107

### Actions

- [x] Tests pipeline locally
  * For such I used a new repo (https://github.com/v1v/test-2fa/tree/feature/synthetics) that contains these changes.
  * https://preview.npmjs.com/package/test-2fa is the package
  * See releases in https://preview.npmjs.com/package/test-2fa/v/0.0.1-alpha.13

### UI

The release process might look like the below steps/screenshots:

- Release from master:

image

- Input approval:

image

Then, there are two UIs (the traditional and the blue ocean one)

image

image

- If approved then

  * Version bump will happen and the tag creation too
  * NPM publish too

image

- If something bad happened before during the release but it might not publsihed anything yet to npm

image

- If something bad happened after it was published to the npmjs registry:

image

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -137,14 +137,31 @@ heartbeat locally
...
+
 If you have access to publish the package to NPM, the process is as follows:
 
 1. Be sure you have checked out the `master` branch and have pulled the latest changes
 1. Run the tests to make sure everything is green
 1. Bump the alpha version by running `npm version prerelease --preid=alpha`
 1. Push commits and tags upstream with `git push upstream master && git push upstream --tags`
 1. Publish to NPM using with `npm publish --tag alpha`
-1. Mark the last published alpha tags as latest using `npm dist-tag add @elastic/synthetics@<$VERSION> latest`
+1. Mark the last published alpha tags as latest using `npm dist-tag add elastic/synthetics@<$VERSION> latest`
```
Comment (v1v) [inline CONTRIBUTING.md:149]: I think there was a typo with the first `@`

**Review by mdelapenya — APPROVED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (v1v) [reply CONTRIBUTING.md]: `@` is required

File: .ci/Jenkinsfile [hunk 2]
```diff
@@ -179,6 +197,75 @@ pipeline {
...
+                withNpmrc(secret: "${env.NPMRC_SECRET}", path: "${env.WORKSPACE}/${env.BASE_DIR}") {
+                  withGitRelease(credentialsId: '2a9602aa-ab9f-4e52-baf3-b71ca88469c7-UserAndToken') {
+                    sh(script: 'npm ci', label: 'prepare the ci environment')
+                    sh(script: 'npm version prerelease --preid=alpha', label: 'bump the alpha version')
+                    gitPush()
+                    gitPush(args: '--tags')
+                    setEnvVar('VERSION', sh(script: "npm view ${env.NPM_PACKAGE} version", returnStdout: true).trim())
+                    log(level: 'INFO', text: "Version to be published '${env.VERSION}'")
+                    withTotpVault(secret: "${env.TOTP_SECRET}", code_var_name: 'TOTP_CODE'){
+                      sh(script: "npm publish --otp=${env.TOTP_CODE} --tag alpha", label: 'publish to NPM')
```
Comment (vigneshshanmugam) [inline .ci/Jenkinsfile:241]: this should already invoke the `npm prepublish` which would clean our files. But just to be extra careful. can you add `npm run clean` during the release step to do a clean release.

**Review by vigneshshanmugam — APPROVED** [review]
Looks good @v1v

File: .ci/Jenkinsfile (hunk 2 above)
Comment (v1v) [reply .ci/Jenkinsfile:241]: There is already a cleanup step earlier -> https://github.com/elastic/synthetics/pull/213/files#diff-ef8b2f5e4d21d81c0eba4754dea3a3dfc2c67d26284f195165b650cf4aff7beaR229

But let me add it for the shake of mind

File: .ci/Jenkinsfile [hunk 3]
```diff
@@ -179,6 +197,75 @@ pipeline {
...
+          environment {
+            NPM_PACKAGE = '@elastic/synthetics'
+          }
+          steps {
+            cleanup()
+            withNodeEnv(){
+              dir("${BASE_DIR}"){
+                withNpmrc(secret: "${env.NPMRC_SECRET}", path: "${env.WORKSPACE}/${env.BASE_DIR}") {
+                  withGitRelease(credentialsId: '2a9602aa-ab9f-4e52-baf3-b71ca88469c7-UserAndToken') {
+                    sh(script: 'npm ci', label: 'prepare the ci environment')
```
Comment (v1v) [inline .ci/Jenkinsfile:234]:
```suggestion
                    sh(script: 'npm run clean', label: 'clean release')
                    sh(script: 'npm ci', label: 'prepare the ci environment')
```

File: .ci/Jenkinsfile (hunk 2 above)
Comment (v1v) [reply .ci/Jenkinsfile:242]: https://github.com/elastic/synthetics/pull/213/files#diff-ef8b2f5e4d21d81c0eba4754dea3a3dfc2c67d26284f195165b650cf4aff7beaR234 is now the clean step pre-release

**Pull request merged**